app/celery/scheduled_tasks.py

```diff
@@ -143,15 +145,57 @@ def switch_current_sms_provider_on_slow_delivery():
                 dao_reduce_sms_provider_priority(provider_name, time_threshold=timedelta(minutes=10))
 
 
+def _check_slow_text_message_delivery_reports_and_raise_error_if_needed(reports: list[SlowProviderDeliveryReport]):
+    total_notifications = sum(report.total_notifications for report in reports)
+    slow_notifications = sum(report.slow_notifications for report in reports)
+    percent_slow_notifications = (slow_notifications / total_notifications) * 100
+
+    # If over 10% of all text messages sent over the period have taken longer than 5 minutes to deliver, let's flag a
+    # sentry error for us to investigate.
+    if percent_slow_notifications >= 10:
+        with sentry_sdk.push_scope() as scope:
+            error_context = {
+                "Dashboard URL": (
+                    "https://grafana-notify.cloudapps.digital/d/icjsQ-MWk2/notify-summary?var-env=preview"
```

Comment:
Discuss: Good work on linking out directly to something to support with the investigation (Grafana). I wonder if we should link to Grafana or if we should link to a runbook entry in our team wiki that explains what an investigator should do (and also links to the Grafana graph too)

Reply:
That might be a better idea, you're right - avoids the grafana URL issue as well. I will set one up and link to it 👍